sympy/assumptions/refine.py

```diff
@@ -291,6 +291,44 @@ def _refine_reim(expr, assumptions):
     return None
 
 
+def refine_sgn(expr, assumptions):
+    """
+    Handler for sign
+
+    Examples
+    ========
+
+    >>> from sympy.assumptions.refine import refine_sgn
+    >>> x = Symbol('x', real = True)
+    >>> expr = sign(x)
+    >>> refine_sgn(expr, Q.positive(x) & Q.nonzero(x))
+    1
+    >>> refine_sgn(expr, Q.negative(x) & Q.nonzero(x))
+    -1
+    >>> refine_sgn(expr, Q.zero(x))
+    0
+    >>> refine_sgn(expr, Q.positive(im(x)))
+    I
+    >>> refine_sgn(expr, Q.negative(im(x)))
+    -I
+    """
+    arg = expr.args[0]
+    if ask(Q.real(arg)):
+        if ask(Q.positive(arg), assumptions):
+            return 1
+        if ask(Q.negative(arg), assumptions):
+            return -1
+        if ask(Q.zero(arg), assumptions):
+            return 0
+    if ask(Q.complex(arg)):
```

Comment:
I think that this definition should be changed to 
```suggestion
    if ask(Q.imaginary(arg)):
```
because SymPy's definition is
```python
>>> sign(I+1).evalf()
0.707106781186548 + 0.707106781186548*I
```

And your definition can be inconsistent
```python
>>> refine_sgn(sign(I+1), True)
I
```